packages/backend/src/api/request.ts

```diff
@@ -171,13 +175,28 @@ export function buildRequest(options: CreateBackendApiOptions) {
         // @ts-expect-error
         status: res?.status,
         statusText: res?.statusText,
+        clerkTraceId: getTraceId(err, res?.headers),
       };
     }
   };
 
   return withLegacyReturn(request);
 }
 
+// Returns either clerk_trace_id if present in response json, otherwise defaults to CF-Ray header
+function getTraceId(data: unknown, headers?: Headers): string | undefined {
+  let traceId: unknown = headers?.get('cf-ray');
+  if (!!data && typeof data === 'object' && 'clerk_trace_id' in data) {
```

Comment:
```suggestion
  if (data && data.clerk_trace_id) {
    return data.clerk_trace_id;
  }
  return '';
```

Reply:
❓ If we want to make it clearer that `clerk_trace_id` from error (non-200) responses takes precedence, you can consider moving line 188 to become the fallback (`cf-ray` is always populated in a response).

Reply:
@SokratisVidros I think we must do the check using the `in` keyword. Otherwise it will not pick it up. 

That is because it is being narrowed from `unknown` -> `object` -> `object with clerk_trace_id in keys`

<img width="916" alt="Screenshot 2023-10-18 at 1 43 29 PM" src="https://github.com/clerkinc/javascript/assets/9081019/591394f4-1d40-4556-a4e7-9d9e4cb1a40f">

Reply:
@georgepsarakis there is a case where we may not even have `cf-ray`. 

If fetch throws an error before making a request, the response will be undefined. This can happen for example if fetch fails due to network issues.